src/app/js/src/SampleApp.jsx

```diff
@@ -1,33 +1,178 @@
+        this.changeName = this.changeName.bind(this);
+        this.addLayer = this.addLayer.bind(this);
+        this.initBaseMap = this.initBaseMap.bind(this);
+        this.createMap = this.createMap.bind(this);
+        this.clearLayers = this.clearLayers.bind(this);
+        this.share = this.share.bind(this);
+    }
+
+    componentDidMount() {
+        this.initBaseMap();
+    }
+
+    initBaseMap() {
+        const baseLayer = new TileLayer({
+            source: new XYZ({
+                url,
+            }),
+        });
+        this.setState({
+            baseLayer,
+            layers: [
+                baseLayer,
+            ],
+        }, this.createMap);
+    }
+
+    createMap() {
+        this.setState({
+            map: new Map({
+                target: 'map',
+                layers: [
+                    this.state.baseLayer,
+                ],
+                view: new View({
+                    center: [0, 0],
+                    zoom: 2,
+                }),
+            }),
+        });
+    }
+
+    changeName(e) {
+        this.setState({
+            name: e.target.value,
+        });
+    }
+
+    changeUrl(e) {
+        this.setState({
+            url: e.target.value,
+        });
+    }
+
+    addLayer() {
+        const newLayer = new TileLayer({
+            source: new XYZ({
+                url: this.state.url,
+            }),
+        });
+        this.state.map.addLayer(newLayer);
+        this.state.layers.push(newLayer);
+        const layerName = (this.state.name === '' ? `Layer ${this.state.layers.length}` : this.state.name);
+        const newTileJSON = {
+            tilejson: '2.2.0',
+            name: layerName,
+            version: '1.0.0',
+            scheme: 'xyz',
+            tiles: [
+                this.state.url,
+            ],
+        };
+        const tileJSONList = this.state.tileJSON;
+        tileJSONList.push(newTileJSON);
+        this.setState({
+            tileJSON: tileJSONList,
+        });
+    }
+
+    clearLayers() {
+        for (let i = 0; i < this.state.layers.length; i += 1) {
+            if (this.state.layers[i] !== this.state.baseLayer) {
+                this.state.map.removeLayer(this.state.layers[i]);
+            }
+        }
+        this.setState({
+            tileJSON: getDefaultTileJSON(),
+            shareLink: '',
+            url: '',
+            name: '',
+        });
+        this.state.layers = [
+            this.state.baseLayer,
+        ];
     }
 
-    inc() {
-        this.props.dispatch(inc());
+    share() {
+        gistRequest.files['tile.json'].content = JSON.stringify(this.state.tileJSON, null, '\t');
+        axios.post('https://api.github.com/gists', gistRequest)
+            .then((response) => {
+                this.setState({
+                    shareLink: `http://bl.ocks.org/d/${response.data.id}/`,
+                });
+            })
+            .catch((err) => {
+                console.log(err);
```

Comment:
Our linting rules flag uses of `console.log`:

```
app_1  | WARNING in ./js/src/SampleApp.jsx
app_1  | (Emitted value instead of an instance of Error)
app_1  | /usr/src/js/src/SampleApp.jsx
app_1  |   147:17  warning  Unexpected console statement  no-console
app_1  |
app_1  | ✖ 1 problem (0 errors, 1 warning)
```

Obviously it's fine for debugging, but I think it's good to get in the habit of not checking uses of it in.